File: app/src/main/java/com/example/dzj/myreader/modle/TxtPager.java
package com.example.dzj.myreader.modle;

import com.google.common.collect.Lists;

import java.util.ArrayList;
import java.util.List;

public class TxtPager implements Cloneable{
    private List<TxtLine> lines;

    public TxtPager(){
        lines = new ArrayList<>();
    }

    public List<TxtLine> getLines() {
        return lines;
    }

    public void setLines(List<TxtLine> lines) {
        this.lines = lines;
    }

    public void addTxtLine(TxtLine line){
        if(lines == null){
            lines = new ArrayList<>();
        }
        lines.add(line);
    }

    public int getSize(){
        return lines.size();
    }

    public TxtLine getLine(int position){
        return lines.get(position);
    }

    @Override
    public TxtPager clone() {
        TxtPager txtPager = null;
        try {
            txtPager = (TxtPager)super.clone();
            txtPager.setLines(Lists.newArrayList(lines));
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
        }
        return txtPager;
    }
}

File: app/src/main/java/com/example/dzj/myreader/modle/TxtPager.kt
package com.example.dzj.myreader.modle

import com.google.common.collect.Lists

class TxtPager : Cloneable {

    var lines : MutableList<TxtLine> = mutableListOf()

    fun addTxtLine(line : TxtLine) {
        lines.add(line)
    }

    fun getLine(position : Int) : TxtLine {
        return lines.get(position)
    }

    public override fun clone(): Any {
        var txtPager: TxtPager? = null
        try {
            txtPager = super.clone() as TxtPager
            txtPager.lines = Lists.newArrayList(lines!!)
        } catch (e: CloneNotSupportedException) {
            e.printStackTrace()
        }

        return txtPager!!
    }
}
